### PR TITLE
[posix] add radio URL parameter fem-lnagain

### PR DESCRIPTION
**Pull request by Moandor-y**
Fixes <https://github.com/openthread/openthread/issues/5787>.

### CODE REVIEW

File: src/posix/platform/radio.cpp [hunk 1]
```diff
@@ -99,6 +99,8 @@ void platformRadioInit(otUrl *aRadioUrl)
     ot::Posix::RadioUrl &radioUrl       = *static_cast<ot::Posix::RadioUrl *>(aRadioUrl);
     bool                 resetRadio     = (radioUrl.GetValue("no-reset") == nullptr);
     bool                 restoreDataset = (radioUrl.GetValue("ncp-dataset") != nullptr);
+    const char *         urlFemLnaGain  = radioUrl.GetValue("fem-lnagain");
```
Comment (bukepo) [inline src/posix/platform/radio.cpp:102]: missing help info for `fem-lnagain`.

File: src/posix/platform/radio.cpp [hunk 2]
```diff
@@ -128,6 +130,16 @@ void platformRadioInit(otUrl *aRadioUrl)
...
+
+    if (urlFemLnaGain != nullptr)
+    {
+        long urlValue = strtol(urlFemLnaGain, nullptr, 0);
+
+        VerifyOrDie(INT8_MIN <= urlValue && urlValue <= INT8_MAX, OT_EXIT_INVALID_ARGUMENTS);
+        femLnaGain = static_cast<int8_t>(urlValue);
+    }
+
+    sRadioSpinel.SetFemLnaGain(femLnaGain);
```
Comment (brianljt) [inline src/posix/platform/radio.cpp]:
```suggestion
        sRadioSpinel.SetFemLnaGain(static_cast<int8_t>(urlValue));
    }
```

By default, LNA gain is set to 0. So I think there is no need to call this if it is not specified in url.

File: src/posix/platform/radio.cpp [hunk 3]
```diff
@@ -99,6 +99,8 @@ void platformRadioInit(otUrl *aRadioUrl)
     ot::Posix::RadioUrl &radioUrl       = *static_cast<ot::Posix::RadioUrl *>(aRadioUrl);
     bool                 resetRadio     = (radioUrl.GetValue("no-reset") == nullptr);
     bool                 restoreDataset = (radioUrl.GetValue("ncp-dataset") != nullptr);
+    const char *         urlFemLnaGain  = radioUrl.GetValue("fem-lnagain");
+    int8_t               femLnaGain     = 0;
```
Comment (brianljt) [inline src/posix/platform/radio.cpp:103]:
```suggestion
```

File: src/posix/platform/radio.cpp (hunk 1 above)
Comment (Moandor-y) [reply src/posix/platform/radio.cpp:102]: Done.

File: src/posix/platform/radio.cpp (hunk 2 above)
Comment (Moandor-y) [reply src/posix/platform/radio.cpp]: Done.

**Review by brianljt — APPROVED** [review]
LGTM!👍 Thanks!

**Review by zhanglongxia — APPROVED** [review]
LGTM

**Review by bukepo — APPROVED** [review]
LGTM:+1:

**Review by jwhui — APPROVED** [review]

**Pull request merged**